### PR TITLE
add readiness probe

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: pkg/probe/probe.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	StatLDSConfigReload = "listener_manager.lds.config_reload"
+	StatCDSConfigReload = "cluster_manager.cds.config_reload"
+)
+
+var (
+	// General stats URI in prometheus format
+	StatsURI = "/stats/prometheus"
+	// Stats endpoint for reading readiness status (usedonly because some stats default to 0)
+	ReadinessURI = fmt.Sprintf("%s?usedonly&filter=(%s|%s)",
+		StatsURI,
```
Comment (yangminzhu) [inline pkg/probe/probe.go:46]: nit: `StatsURI` is only used here, probably just encode it directly in the string.

File: pkg/probe/probe.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		return false, err
+	}
+	// server state must be 0
+	if state, exists := metrics[PromStatName(StatServerState)]; !exists {
+		return false, nil
+	} else if len(state.Metric) != 1 || state.Metric[0].Gauge == nil ||
+		state.Metric[0].Gauge.Value == nil || *state.Metric[0].Gauge.Value != 0 {
+		return false, nil
+	}
+	// workers started must be 1
```
Comment (yangminzhu) [inline pkg/probe/probe.go:77]: it must be 1 because the envoy is just started (no traffic)?

File: pkg/probe/probe.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	// Values described here:
+	// https://www.envoyproxy.io/docs/envoy/v1.15.0/api-v3/admin/v3/server_info.proto#envoy-v3-api-enum-admin-v3-serverinfo-state
+	// LIVE is value 0.
+	StatServerState = "server.state"
+
+	// Boolean indicating whether listeners have been initialized on the workers.
+	StatListenerManagerWorkersStarted = "listener_manager.workers_started"
+
+	// Total number of updates that resulted in a config reload.
+	StatLDSConfigReload = "listener_manager.lds.config_reload"
```
Comment (yangminzhu) [inline pkg/probe/probe.go:37]: These seems not used?

File: pkg/probe/probe.go (hunk 1 above)
Comment (kyessenov) [reply pkg/probe/probe.go:46]: I was thinking of extending it with other stat queries later on.

File: pkg/probe/probe.go (hunk 2 above)
Comment (kyessenov) [reply pkg/probe/probe.go:77]: 1 means "true", so the workers have started.

**Pull request closed without merging**